app/javascript/mastodon/components/status_list.js

```diff
@@ -41,16 +42,26 @@ class StatusList extends ImmutablePureComponent {
     } else if (this.props.onScroll) {
       this.props.onScroll();
     }
-  }
+  }, 200, {
+    trailing: true,
+  });
```

Comment:
Looks like TIMELINE_SCROLL_TOP is [already debounced elsewhere](https://github.com/tootsuite/mastodon/blob/37dbfa4cd7f97cfbab68d1bfd774aabf5ca2e0af/app/javascript/mastodon/features/ui/containers/status_list_container.js#L53-L64)?

As for delaying fetch requests by 200ms, that doesn't seem like a big deal to me; I was more worried about continuously setting `scrollTop`while undebounced and thus invoking layout continuously.